src/ngp_algorithms/SDRWallFuncAlg.C

```diff
@@ -108,14 +113,20 @@ void SDRWallFuncAlg<BcAlgTraits>::execute()
         const int nodeR = meSCS->ipNodeMap(fdata.faceOrd)[ip];
         const int nodeL = meSCS->opposingNodes(fdata.faceOrd, ip);
 
-        DoubleType ypBip = 0.0;
-        for (int d=0; d < BcAlgTraits::nDim_; ++d) {
-          const DoubleType nj = v_area(ip, d) / aMag;
-          const DoubleType ej = 0.25 * (v_coord(nodeR, d) - v_coord(nodeL, d));
-          ypBip += nj * ej * nj * ej;
+        DoubleType ypBip;
+        if (RANSAblBcApproach_) {
+          // set ypBip to roughness height for wall function calculation
+          ypBip = z0_;
```

Comment:
Accessing instance member variables from device kernels will fail on GPUs.

Reply:
Could you explain this issue more and give an example from the code of how to work around this issue? 

Is the problem that I can't use `RANSAblBcApproach_`, that I'm not allowed to create a variable such as `ypBip`, that I'm not allowed to have if/else statements, or something else? 

Thank you.

Reply:
You can't use `RANSAblBcApproach_` and `z0_` inside a `run_{entity, elem, face_elem}_algorithm` loop you'll have to create local copies within the execute function and capture that with the KOKKOS_LAMBDA. You can see how we have done this for other class instance variables

https://github.com/Exawind/nalu-wind/blob/5ce9f628676cdf7217330a3a2c832abfcdeb294f/src/ngp_algorithms/SDRWallFuncAlg.C#L77-L85